lib/HunkView.ml

```diff
@@ -124,14 +153,22 @@ let create_summary (start_line_num : int) (hunk_length : int)
       (Printf.sprintf "@@ %s%d,%d @@" sign start_line_num hunk_length)
   else W.string ~attr (Printf.sprintf "@@ %s0,0 @@" sign)
 
-let ui_of_hunk_side_by_side (hunk : string Patch.hunk) : Nottui.ui =
-  let attr_mine = Notty.A.(fg red ++ st bold) in
-  let attr_their = Notty.A.(fg green ++ st bold) in
+let ui_of_hunk_side_by_side (hunk : string Patch.hunk)
+    (rendering_mode : Types.rendering_mode) : Nottui.ui =
+  let attr_mine, attr_their =
+    match rendering_mode with
+    | Types.Color -> (Notty.A.(fg red ++ st bold), Notty.A.(fg green ++ st bold))
+    | Types.TextMarkers -> (Notty.A.empty, Notty.A.empty)
```

Comment:
`rendering_mode` is matched in too many places. For example, here it's matched to know what to pass to `lines_with_numbers`, which match it again for other reasons.
Also, it's matched several times in `ui_hunk_summary` to decides whether or not to apply attributes, which is also what this match does.